reactioncommerce/reaction
# Properly create/remove/prune shipping records everywhere

Closes #2836 and #2870 #2838 (and probably a bunch of other bugs)

### To Test

1. Create a shop
1. Enable Stripe Connect
1. As the shop admin, add a product
1. Enable flat rate shipping
1. In your original browser, add only your newly created product
1. Checkout
1. Observe that shipping methods are properly displayed
1. Checkout
1. Observe there is no error

That's just the basic test case for the bugs, but really this PR became "make shipping work in marketplace" so it needs to get tested with all the various scenarios e.g.:

1. Single shop with products for that shop
1. Multi shop with just products for the second shop
1. Multi-shop with products for both primary and merchant shop
1. Shippo
1. Flat-rate
